channels/rdpdr/server/rdpdr_main.c

```diff
@@ -134,8 +184,13 @@ static int rdpdr_server_read_general_capability_set(RdpdrServerContext* context,
 	Stream_Read_UINT32(s, extendedPdu); /* extendedPdu (4 bytes) */
 	Stream_Read_UINT32(s, extraFlags1); /* extraFlags1 (4 bytes) */
 	Stream_Seek_UINT32(s); /* extraFlags2 (4 bytes), must be set to zero, reserved for future use */
-	Stream_Read_UINT32(s, SpecialTypeDeviceCap); /* SpecialTypeDeviceCap (4 bytes) */
+	if (header->Version == GENERAL_CAPABILITY_VERSION_02)
+	{
+		Stream_Read_UINT32(s, SpecialTypeDeviceCap); /* SpecialTypeDeviceCap (4 bytes) */
+	}
```

Comment:
I'm not sure here but is the SpecialTypeDeviceCap optional and only sent if GENERAL_CAPABILITY_VERSION_02 is used?